### PR TITLE
Add POST /api/propagation-credentials

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes a part of #1965 
This branch was created from `1965-retrieve-credentials` (#2063), rebased on develop (conflicts were resolved), and then extended. Merge after #2063. 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] ~Have you successfully tested your changes locally? Elaborate:~
    > ~Tested by {Running the Monkey locally with relevant config/running Island/...}~
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/monkey_island/cc/resources/propagation_credentials.py [hunk 1]
```diff
@@ -21,3 +24,54 @@ def get(self):
...
+                    self._credentials_repository.save_configured_credentials(credentials_object)
+            except StorageError as err:
+                return make_response(
+                    {"error": f"Error encountered while storing credentials: {err}"}, 400
+                )
+
+        return make_response({}, 200)
+
+    @staticmethod
+    def _create_credentials_object(credentials: dict) -> Credentials:
```
Comment (ilija-lazoroski) [inline monkey/monkey_island/cc/resources/propagation_credentials.py:57]: This will go away when we have the serialization ready. Mark it with a note.

**Review by ilija-lazoroski — CHANGES_REQUESTED** [review]
It needs rebase on the `GET` branch, UT and a note to remove the creation of credentials object.

File: monkey/monkey_island/cc/resources/propagation_credentials.py [hunk 2]
```diff
@@ -1,15 +1,83 @@
...
+        self._credentials_repository = credentials_repository
+
     def get(self):
-        config = ConfigService.get_flat_config(should_decrypt=True)
+        propagation_credentials = self._credentials_repository.get_all_credentials()
+
+        return make_response({"propagation_credentials": propagation_credentials})
+
+    def post(self):
+        body = json.loads(request.data)
```
Comment (ilija-lazoroski) [inline monkey/monkey_island/cc/resources/propagation_credentials.py:33]: I think we can use `request.json` here.

File: monkey/monkey_island/cc/resources/propagation_credentials.py [hunk 3]
```diff
@@ -1,15 +1,83 @@
...
-from monkey_island.cc.services.config import ConfigService
+
+
+class UnknownCredentialTypeError(Exception):
+    pass
+
+
+class PropagationCredentialType(Enum):
+    STOLEN = "stolen"
+    CONFIGURED = "configured"
```
[the remark this answers is not in the document]
Comment (mssalvatore) [reply monkey/monkey_island/cc/resources/propagation_credentials.py:18]: We discussed this in planning and decided against it.

File: monkey/monkey_island/cc/resources/propagation_credentials.py [hunk 4]
```diff
@@ -1,15 +1,83 @@
...
+                credentials_object = PropagationCredentials._create_credentials_object(credentials)
+            except Exception as exc:
+                return make_response({"error": f"Couldn't create 'Credentials' object: {exc}"}, 400)
+
+            try:
+                if body.get("type") == PropagationCredentialType.STOLEN:
+                    self._credentials_repository.save_stolen_credentials(credentials_object)
+                elif body.get("type") == PropagationCredentialType.CONFIGURED:
+                    self._credentials_repository.save_configured_credentials(credentials_object)
+                else:
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/resources/propagation_credentials.py:47]: These need to reset before they save.

File: monkey/monkey_island/cc/resources/propagation_credentials.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shreyamalviya) [reply monkey/monkey_island/cc/resources/propagation_credentials.py:18]: And then we discussed again and decided to do it.

**Pull request closed without merging**